### DISCUSSION TITLE
how can i convert a file with ./gradlew build distZip command

### DISCUSSION THREAD

**Discussion by ikuruzum** [dotlin-org/dotlin#4] [Q&A] [answered]
i want to convert a simple kotlin library to dart library for using it's functions. so how can i do that. idk anything about kotlin so any detailed information appreciated

**Comment by wilkomanger** [comment, marked as answer]
Hi! The `. /gradlew build distZip` is used to compile the Dotlin compiler itself. Then you will have a binary you can use (inside the zip file). However, it's still in heavy development and for now only useful to compile some loose files.

I'm currently working on a big change where the Dart project structure is used, from then on it will be better and better for compiling full projects.

Another thing: Dotlin aims to bring Kotlin as a language to the Dart ecosystem. This means that you can't compile any Kotlin/JVM or Kotlin multiplatform project, but you _will_ be able to use any Dart dependency.

I might write a tool for doing a sort of "initial conversion" from a e.g. Kotlin/JVM project to a Dotlin project, but we'll have to see.

So yeah, for now you can't do what you asked. Hope that helps! 

Just for my own curiosity, which library?

**Comment by ikuruzum** [comment]
https://github.com/cosinekitty/astronomy this one, its written in more languages ​​besides kotlin but not in dart. I've looked at converters for other languages ​​as well, but they're all either deprecated or obsolete. as a last resort i will convert it using ChatGPT, I can't think of a different method right now.

**Reply by wilkomanger** [reply]
I did find [solar_calculator](https://pub.dev/packages/solar_calculator), not sure if it does what you need, though. Anyway, good luck!
